pkg/frontend/middleware/validate.go

```diff
@@ -48,50 +48,50 @@ func Validate(env env.Interface, apis map[string]*api.Version) func(http.Handler
 			if _, found := vars["subscriptionId"]; found {
 				_, err := uuid.FromString(vars["subscriptionId"])
 				if err != nil {
-					api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeInvalidSubscriptionID, "", "The provided subscription identifier '%s' is malformed or invalid.", vars["subscriptionId"])
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidSubscriptionID, "", "The provided subscription identifier '%s' is malformed or invalid.", vars["subscriptionId"])
```

Comment:
if this is a call for get the subscription it should be a 404.  I'm less sure on other calls @amanohar  thoughts?

Reply:
ditto - validation the subscriptionId is a uuid